mimic/rest/loadbalancer_api.py

```diff
@@ -54,10 +55,128 @@ def resource_for_region(self, region, uri_prefix, session_store):
         return lb_region.app.resource()
 
+    def _get_session(self, session_store, tenant_id):
+        """
+        Retrieve or create a new LoadBalancer session from a given tenant identifier
+        and :obj:`SessionStore`.
+
+        For use with ``data_for_api``.
+
+        Temporary hack; see this issue
+        https://github.com/rackerlabs/mimic/issues/158
+        """
+        return (
+            session_store.session_for_tenant_id(tenant_id)
+            .data_for_api(self, lambda: GlobalCLBCollections(
+                tenant_id=tenant_id,
+                clock=session_store.clock
+            ))
+        )
+
+
+@implementer(IAPIMock, IPlugin)
+@attributes(["lb_api"])
+class LoadBalancerControlApi(object):
+    """
+    This class registers the load balancer controller API in the service
+    catalog.  It does NOT implement individual endpoints.
+    """
+    def catalog_entries(self, tenant_id):
+        """
+        Cloud load balancer controller endpoints.
+        """
+        return [
+            Entry(
+                tenant_id, "rax:load-balancer", "cloudLoadBalancerControl",
+                [
+                    Endpoint(tenant_id, region, text_type(uuid4()), prefix="v2")
+                    for region in self.lb_api._regions
+                ]
+            )
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Get an :obj:`twisted.web.iweb.IResource` for the given URI prefix;
+        implement :obj:`IAPIMock`.
+        """
+        lbc_region = LoadBalancerControlRegion(api_mock=self, uri_prefix=uri_prefix,
+                                               session_store=self.lb_api.session_store, region=region)
+        return lbc_region.app.resource()
+
+
+@attributes(["api_mock", "uri_prefix", "session_store", "region"])
+class LoadBalancerControlRegion(object):
+    """
+    Klein routes for load balancer's control API within a particular region.
+    """
+
+    app = MimicApp()
+
+    def session(self, tenant_id):
+        """
+        Gets a session for a particular tenant, creating one if there isn't
+        one.
+        """
+        tenant_session = self._session_store.session_for_tenant_id(tenant_id)
+        clb_global_collection = tenant_session.data_for_api(
+            self._api_mock,
+            lambda: GlobalCLBCollections(
+                tenant_id=tenant_id,
+                clock=self._session_store.clock))
+        clb_region_collection = clb_global_collection.collection_for_region(
+            self.region_name)
+        return clb_region_collection
+
+    def _collection_from_tenant(self, tenant_id):
+        """
+        Retrieve the server collection for this region for the given tenant.
+        """
+        return (self.api_mock.lb_api._get_session(self.session_store, tenant_id)
+                .collection_for_region(self.region))
+
+    @app.route('/v2/<string:tenant_id>/loadbalancer/<int:clb_id>/setAttr', methods=['POST'])
```

Comment:
Using `setAttr` as an endpoint here is unfortunately inconsistent with the noun/verb structure of Mimic APIs (and generally things which claim to be REST, but I'll spare you my rant about REST for another day).

URIs should generally identify a noun, and the verb you're executing should be identified by a verb.  There are all kinds of silly reasons that people use to justify this about "state transfers" and "read operation idempotence" and "cache friendliness" which don't really have much value (oops that REST rant leaked through a little).  However, there is a very real reason that you want to follow that pattern: namespace pollution.  The control plane may have subsidiary components of a load-balancer that we want to expose - like, for example, the attributes themselves - which might have their own identifiers.  So `/loadbalancer/<clb_id>/setAttr` masks a potential future noun endpoint called `setAttr`.

The Nova control endpoint's equivalent API is under `/attributes/` and the equivalent verb is a POST.  It's done that way to facilitate bulk updates, and the noun involved is the *attribute* rather than the server because it makes it easier to simulate data-center conditions like a whole rack failing at once, which is something that might happen with load balancers as well.

I think mirroring that structure as closely as possible would make this easier to use for people learning Mimic.  However, it's possible that we would want to support multiple styles of this for Nova, which is why the present control endpoint is under `/attributes/`, to preserve the `/servers/` namespace for individual-server inspection and update functionality.

So, in keeping with the "noun" being operated on here being "attributes" and the operation being "change some of them", I think this endpoint should live at `/v2/<string:tenant_id>/loadbalancer/<int:clb_id>/attributes/`.  `POST` is a good catch-all "do stuff" HTTP verb, and it's what we used in the Nova stuff so it's probably fine.  (Putting on my protocols wonk hat for a moment though, really the *correct* correct verb here would be [`PATCH`](http://datatracker.ietf.org/doc/rfc5789/?include_text=1) because this is a partial update to an existing object which should always fail if the object being identified doesn't exist already.)